src/upload-button.tsx

```diff
@@ -0,0 +1,384 @@
+        >
+            <p>
+                {cockpit.format(
+                    _("A file with the same name already exists in \"$0\". Replacing it will overwrite its content."),
+                    path.join('/')
+                )}
+            </p>
+            <Flex
+              spaceItems={{ default: "spaceItems3xl" }}
+              className="conflict-modal-files"
+            >
+                <FlexItem>
+                    <b>{_("New file")}</b>
+                    <p>{cockpit.format_bytes(uploadFile.size)}</p>
+                    <p>{timeformat.dateTime(uploadFile.lastModified)}</p>
+                </FlexItem>
+                <FlexItem>
+                    <b>{_("Original file on server")}</b>
+                    <p>{cockpit.format_bytes(file.size)}</p>
+                    <p>{timeformat.dateTime(file.mtime * 1000)}</p>
+                </FlexItem>
+            </Flex>
+            {showApplyAll &&
+            <Checkbox
+              id="replace-all"
+              label={_("Apply this action to all conflicting files")}
+              isChecked={applyToAll}
+              onChange={() => setApplyToAll(!applyToAll)}
+            />}
+        </Modal>
+    );
+};
+
+export const UploadButton = ({
+    files,
+    path,
+} : {
+    files: FolderFileInfo[],
+    path: string[],
+}) => {
+    const ref = useRef<HTMLInputElement>(null);
+    const { addAlert } = useContext(FilesContext);
+    const Dialogs = useDialogs();
+    const [showPopover, setPopover] = React.useState(false);
+    const [uploadedFiles, setUploadedFiles] = useState<{[name: string]:
+                                                        {file: File, progress: number, cancel:() => void}}>({});
+
+    const handleClick = () => {
+        if (ref.current) {
+            ref.current.click();
+        }
+    };
+
+    const beforeUnloadHandler = (event: BeforeUnloadEvent) => {
+        event.preventDefault();
+
+        // Included for legacy support, e.g. Chrome/Edge < 119
+        event.returnValue = true;
+    };
+
+    const onUpload = async (event: React.ChangeEvent<HTMLInputElement>) => {
+        cockpit.assert(event.target.files, "not an <input type='file'>?");
+        let next_progress = 0;
+        const toUploadFiles = [];
+
+        const resetInput = () => {
+        // Reset input field in the case a download was cancelled and has to be re-uploaded
+        // https://stackoverflow.com/questions/26634616/filereader-upload-same-file-again-not-working
+            event.target.value = "";
+        };
+
+        // Before uploading, ask the user if any existing files should be overwritten or skipped.
+        // Inefficient....
+        let resolution;
+        let applyToAll = false;
+        for (let i = 0; i < event.target.files.length; i++) {
+            const inputFile = event.target.files[i];
+            console.log(inputFile, resolution);
+
+            if (resolution && applyToAll && resolution.skip)
+                break;
```

Comment:
Is `break` right from a UI standpoint?  We want to only skip all the files that already exist, right?  Not all files...

Reply:
We have a checkbox which says "apply action to all files" so if you click skip this skips uploading all files.

Reply:
Hmm, that doesn't feel right. I'd expect to only skip *existing* files, not *all* files. In the latter case, this would just be a plain "cancel", and there's already a button for that.